### PR TITLE
chore: split and move `utils` to `cargo_near_build`

### PR DESCRIPTION
**Pull request by dj8yfo**
- **chore: move cargo_near::util::print -> cargo_near_build::pretty_print**
- **chore: move `util::{copy, force_cononicalize_dir}` to `cargo_near_build::fs`**
- **chore: move `wasm32_exists` logic to `cargo_near_build::cargo_native::target`**
- **chore: move `dylib_extention` to `cargo_near_build::cargo_native`**
- **chore: move `util::{...near types...}` -> `cargo_near_build::types::near::{...}`**
- **chore: parameterize `CompilationArtifact` with `T: ArtifactType`**
- **chore: move `util::{... build funcs ...}` -> `cargo_near_build::cargo_native::compile::run`**
- **chore: move `util::extract_abi_entries` -> `cargo_near_build::near::abi`**

### CODE REVIEW

**Comment by dj8yfo** [comment]
@race-of-sloths pong

**Comment by race-of-sloths** [comment]
@dj8yfo Thank you for your contribution! Your pull request is now a part of the [Race of Sloths](https://race-of-sloths.com)!
Great job! Usain Bolt jealous of your results!

[<picture>
    <source media="(prefers-color-scheme: dark)" srcset="https://badge.race-of-sloths.com/dj8yfo?type=bot&theme=dark">
    <source media="(prefers-color-scheme: light)" srcset="https://badge.race-of-sloths.com/dj8yfo?type=bot&theme=light">
    Shows profile picture for the author of the PR
</picture>](https://race-of-sloths.com/profile/dj8yfo)


<details><summary>Current status: <i>waiting for scoring</i></summary>

We're waiting for maintainer to score this pull request with `@race-of-sloths score [0,1,2,3,5,8,13]` command. Alternatively, autoscoring [1,2] will be applied for this pull request

</details><details><summary>What is the Race of Sloths</summary>
<p>
Race of Sloths is a friendly competition where you can participate in challenges and compete with other open-source contributors within your normal workflow

For contributors:

- Tag @race-of-sloths inside your pull requests
- Wait for the maintainer to review and score your pull request
- Check out your position in the [Leaderboard](https://race-of-sloths.com/leaderboard)
- Keep weekly and monthly streaks to reach higher positions
- Boast your contributions with a dynamic picture of your [Profile](https://race-of-sloths.com/profile/dj8yfo)

For maintainers:
- Score pull requests that participate in the Race of Sloths
- Engage contributors with fair scoring and fast responses so they keep their streaks
- Promote the Race to the point where the Race starts promoting you
- Grow the community of your contributors

Feel free to check [our website](https://race-of-sloths.com) for additional details!
</p>
</details> 

<details>
<summary>Bot commands</summary>

- For contributors
  - **Include a PR:** `@race-of-sloths include` to enter the Race with your PR
- For maintainers:
  - **Invite contributor** `@race-of-sloths invite` to invite the contributor to participate in a race or include it, if it's already a runner.
  - **Assign points:** `@race-of-sloths score [1/2/3/5/8/13]` to award points based on your assessment.
  - **Reject this PR:** `@race-of-sloths exclude` to send this PR back to the drawing board.
  - **Exclude repo:** `@race-of-sloths pause` to stop bot activity in this repo until `@race-of-sloths unpause` command is called

</details>

**Review by frol — APPROVED** [review]
@race-of-sloths score 13

**Pull request merged**
